### PR TITLE
Add unary plus/negation feature plus test coverage

### PR DESCRIPTION
**Pull request by hokevins**
Hi, following up on Issue #72 , I am making a Pull Request for `react-jsx-parser` to support unary plus/negation.  Created unit tests, all tests pass, and ESLint passes as well.  I did not commit the `react-jsx-parser.min.js` file though as I wasn't sure about your project repo's build/sem-ver workflow.  If you could please build/deploy/bump the to the latest version after approval, that would be much appreciated.  Please kindly approve and/or let me know next steps to proceed with this PR.

Thank you.

### CODE REVIEW

**Comment by hokevins** [comment]
Oh that's strange, all tests passed locally when I ran `yarn test` in my terminal, but the circle/ci is reporting it as failed as above.  Is it running the tests against the source folder or the output minified file?

File: source/components/JsxParser.js [hunk 1]
```diff
@@ -89,6 +89,15 @@ export default class JsxParser extends Component {
           default:
             return undefined
         }
+      case 'UnaryExpression':
+        switch (expression.operator) {
+          case '+':
+            return expression.argument.value
+          case '-':
+            return -1 * expression.argument.value
+          default:
```
Comment (TroyAlford) [inline source/components/JsxParser.js]: I think we should also support the `!` operator, which also falls under `UnaryExpression` in Acorn.
```suggestion
          case '!':
            return !expression.argument.value
          default:
```

File: source/components/JsxParser.test.js [hunk 2]
```diff
@@ -698,6 +698,16 @@ describe('JsxParser Component', () => {
...
+      const { rendered, component } = render(<JsxParser jsx={'<span testProp={+60}>{ +75 }</span>'} />)
+      expect(rendered.childNodes[0].textContent).toEqual('75')
+      expect(component.ParsedChildren[0].props.testProp).toEqual(60)
+    })
+    it('can execute unary negation operations', () => {
+      const { rendered, component } = render(<JsxParser jsx={'<span testProp={-60}>{ -75 }</span>'} />)
+      expect(rendered.childNodes[0].textContent).toEqual('-75')
+      expect(component.ParsedChildren[0].props.testProp).toEqual(-60)
+    })
   })
```
Comment (TroyAlford) [inline source/components/JsxParser.test.js]:
```suggestion
    it('can execute unary NOT operations', () => {
      const { rendered, component } = render(<JsxParser jsx={'<span testProp={!60}>{ !false }</span>'} />)
      expect(rendered.childNodes[0].textContent).toEqual('true')
      expect(component.ParsedChildren[0].props.testProp).toEqual(false)
    })
  })
```

File: source/components/JsxParser.js (hunk 1 above)
Comment (hokevins) [reply source/components/JsxParser.js]: This won't work and will fail the tests, you must also coerce T/F into strings to render and then back into T/F for props.  I will update with an additional commit to accommodate this feature request and pass all tests.

**Comment by hokevins** [comment]
Hi, your comments have been addressed, so I've added 2 additional commits to include feature support for unary NOT operations and tests to cover that as well.  Thank you.  Please kindly let me know next steps?

File: source/components/JsxParser.js (hunk 1 above)
Comment (hokevins) [reply source/components/JsxParser.js]: Addressed in newest commit:  Update to include unary NOT operations support.

This suggested solution actually will not work, must instead coerce the T/F values to a string in order to be rendered, and then back to a T/F for props.

**Comment by hokevins** [comment]
Hi @TroyAlford , circling back to this PR currently awaiting your review.  Thank you.

**Pull request merged**